### PR TITLE
Presence: get clientId from Auth, not options directly

### PR DESCRIPTION
**Pull request by SimonWoolf**
Fixes https://github.com/ably/ably-js/issues/198

### CODE REVIEW

**Comment by mattheworiordan** [comment]
LGTM

File: common/lib/client/realtimepresence.js [hunk 1]
```diff
@@ -102,7 +104,7 @@ var RealtimePresence = (function() {
 	};
 
 	RealtimePresence.prototype.leave = function(data, callback) {
-		if(!this.clientId)
+		if(!getClientId(this))
 			throw new Error('clientId must have been specified to enter or leave a presence channel');
```
Comment (paddybyers) [inline common/lib/client/realtimepresence.js]: ErrorInfo?

**Comment by paddybyers** [comment]
One comment but otherwise lgtm.

Thanks

File: common/lib/client/realtimepresence.js (hunk 1 above)
Comment (SimonWoolf) [inline common/lib/client/realtimepresence.js]: https://github.com/ably/ably-js/commit/ea53db996db520e01d7ced261d42f85200392462

**Comment by paddybyers** [comment]
@SimonWoolf: what happened to this?

**Comment by SimonWoolf** [comment]
Not much, just needed rebasing

**Comment by paddybyers** [comment]
Thanks

**Pull request merged**